java/client/src/main/java/glide/api/models/GlideString.java

```diff
@@ -4,26 +4,43 @@
 import java.nio.charset.StandardCharsets;
 import java.util.Arrays;
 import java.util.concurrent.atomic.AtomicBoolean;
+import lombok.EqualsAndHashCode;
 import lombok.Getter;
 
-// TODO docs for the god of docs
+/**
+ * Represents a Valkey string type. Since Valkey stores strings as <code>byte[]</code>, such strings
+ * can contain non-UTF8 compatible symbols or even arbitrary binary data BLOBs.<br>
+ * This class stores data <code>byte[]</code> too, but provides API to represent data as a {@link
+ * String} if conversion is possible.
+ */
+@EqualsAndHashCode
 public class GlideString implements Comparable<GlideString> {
 
+    /** The data itself. */
     @Getter private byte[] bytes;
-    private String string = null;
+
+    /**
+     * Stores a string when it is possible.<br>
+     * Value is written if conversion requested via {@link #toString()} or {@link #getString()} or
+     * checked with {@link #canConvertToString()} only if conversion is possible.
+     */
+    @EqualsAndHashCode.Exclude private String string = null;
 
     /** Flag whether possibility to convert to string was checked. */
     private final AtomicBoolean conversionChecked = new AtomicBoolean(false);
 
+    /** Constructor is private - use {@link #gs} or {@link #of} to instantiate an object. */
     private GlideString() {}
```

Comment:
update constructor to take String and byte[]
😭

Reply:
and make bytes and string final

Reply:
String cannot be final, becase it is determined on demand only. Or not, if data isn't string-compatible.